i2c/pcf8523_i2c/pcf8523_i2c.c

```diff
@@ -0,0 +1,162 @@
+
+#ifdef i2c_default
+static void pcf8520_reset() {
+    // Two byte reset. First byte register, second byte data
+    // There are a load more options to set up the device in different ways that could be added here
+    uint8_t buf[] = {0x00, 0x58};
+    i2c_write_blocking(i2c_default, addr, buf, 2, false);
+}
+
+static void pcf820_write_current_time() {
+    // buf[0] is the register to write to
+    // buf[1] is the value that will be written to the register
+    uint8_t buf[2];
+
+    //Write values for the current time in the array
+    //index 0 -> second: bits 4-6 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 1 -> minute: bits 4-6 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 2 -> hour: bits 4-5 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 3 -> day of the month: bits 4-5 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 4 -> day of the week: where Sunday = 0x00, Monday = 0x01, Tuesday... ...Saturday = 0x06
+    //index 5 -> month: bit 4 is responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 6 -> year: bits 4-7 are responsible for the ten's digit and bits 0-3 for the unit's digit
+
+    //NOTE: if the value in the year register is a multiple for 4, it will be considered a leap year and hence will include the 29th of February
+
+    uint8_t current_val[7] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+    for (int i = 3; i < 10; ++i){
+        buf[0] = i;
+        buf[1] = current_val[i - 3];
+        i2c_write_blocking(i2c_default, addr, buf, 2, false);
+    }
+}
+
+static void pcf8520_read_raw(uint8_t *buffer) {
+    // For this particular device, we send the device the register we want to read
+    // first, then subsequently read from the device. The register is auto incrementing
+    // so we don't need to keep sending the register we want, just the first.
+
+    // Start reading acceleration registers from register 0x3B for 6 bytes
+    uint8_t val = 0x03;
+    i2c_write_blocking(i2c_default, addr, &val, 1, true); // true to keep master control of bus
+    i2c_read_blocking(i2c_default, addr, buffer, 7, false);
+}
+
+
+void pcf8520_set_alarm(){
+    // buf[0] is the register to write to
+    // buf[1] is the value that will be written to the register
+    uint8_t buf[2];
+
+    // Default value of alarm register is 0x80
+    // Set bit 8 of values to 0 to activate that particular alarm
+    // Index 0 -> minute: bits 4-5 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    // Index 1 -> hour: bits 4-6 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    // Index 2 -> day of the month: bits 4-5 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    // Index 3 -> day of the week: where Sunday = 0x00, Monday = 0x01, Tuesday... ...Saturday = 0x06
+
+    uint8_t alarm_val[4] = {0x01, 0x80, 0x80, 0x80};
```

Comment:
A comment saying what "actual time" this alarm_val corresponds to would be nice :slightly_smiling_face: